src/topology/algebra/uniform_convergence.lean

```diff
@@ -78,4 +109,115 @@ begin
           uniform_convergence_on.uniform_equiv_prod_arrow.symm.uniform_continuous⟩
 end
 
+@[to_additive]
+protected lemma uniform_convergence_on.has_basis_nhds_one_of_basis (𝔖 : set $ set α)
+  (h𝔖₁ : 𝔖.nonempty) (h𝔖₂ : directed_on (⊆) 𝔖) {p : ι → Prop}
+  {b : ι → set G} (h : (𝓝 1 : filter G).has_basis p b) :
+  (@nhds (α → G) (uniform_convergence_on.topological_space α G 𝔖) 1).has_basis
```

Comment:
I've basically been hesitating for the last six months about making a type alias for "functions with the topology of uniform convergence (on some sets)". 

The thing is I'm actually not sure about how much these things are going to be used on bare functions (I mean as a topology, because we also have [tendsto_uniformly](https://leanprover-community.github.io/mathlib_docs/topology/uniform_space/uniform_convergence.html#tendsto_uniformly) and variants). All cases I can think of where we put this topology on something, it will be on a separated type anyway (such as (bounded) continuous functions or continuous linear maps). So maybe we just deal with `@`s everywhere when doing the general theory on bare functions, but when it's applied to more specialized types everything will work just fine. That said, it turns out that the bare function case is also useful for talking about equicontinuity (see #16467) so I'm not sure I agree with myself anymore :sweat_smile: 

Besides, I'm not even sure adding a type alias would help that much. I remember seeing cases where type ascription to a defeq type were actually completely ignored by Lean when doing instance search. But I can try it if you want.

Reply:
Let me think about this over the weekend and get back to you.